internal/provider/resource_silo.go

```diff
@@ -0,0 +1,472 @@
+				Description: "Whether this silo is present in the silo_list output.",
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplaceIfConfigured(),
+				},
+			},
+			"mapped_fleet_roles": schema.MapAttribute{
+				Optional:    true,
+				Description: "Mapped Fleet Roles for the Silo.",
+				ElementType: types.ListType{ElemType: types.StringType},
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Name of the silo.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"quotas": schema.SingleNestedAttribute{
+				Required:    true,
+				Description: "Limits the amount of provisionable CPU, memory, and storage in the Silo.",
+				Attributes: map[string]schema.Attribute{
+					"cpus": schema.Int64Attribute{
+						Required:    true,
+						Description: "Amount of virtual CPUs available for running instances in the Silo.",
+					},
+					"memory": schema.Int64Attribute{
+						Required:    true,
+						Description: "Amount of RAM (in bytes) available for running instances in the Silo.",
+					},
+					"storage": schema.Int64Attribute{
+						Required:    true,
+						Description: "Amount of storage (in bytes) available for disks or snapshots.",
+					},
+				},
+			},
+			"tls_certificates": schema.ListNestedAttribute{
+				Required:    true,
+				Description: "Initial TLS certificates to be used for the new Silo's console and API endpoints.",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"cert": schema.StringAttribute{
+							Description: "PEM-formatted string containing public certificate chain.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+						"description": schema.StringAttribute{
+							Description: "Certificate description.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+						"key": schema.StringAttribute{
+							Description: "PEM-formatted string containing private key.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+						"name": schema.StringAttribute{
+							Description: "Name associated to the certificate.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+							Validators: []validator.String{
+								stringvalidator.RegexMatches(regexp.MustCompile(`^[a-zA-Z0-9-]+$`),
+									"allowed characters are lowercase ASCII, digits, and \"-\""),
+							},
+						},
+						"service": schema.StringAttribute{
+							Description: "Service using this certificate.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
```

Comment:
We can update this to validate that the value is exactly `external_api`. Alternatively we can mark it optional with a default value so the user doesn't even need to set it. Probably the former since that's in line with the API.